rasa/core/policies/memoization.py

```diff
@@ -279,25 +280,43 @@ def _back_to_the_future_again(tracker) -> Optional[DialogueStateTracker]:
                     idx_of_second_action = e_i
                     break
 
-        if idx_of_second_action is None:
-            return None
+        # use first action, if we went first time and second action, if we went again
+        idx_to_use = idx_of_second_action if again else idx_of_first_action
+        if idx_to_use is None:
+            return
+
         # make second ActionExecuted the first one
-        events = tracker.applied_events()[idx_of_second_action:]
+        events = applied_events[idx_to_use:]
         if not events:
-            return None
+            return
```

Comment:
nit: `return None` is actually preferable as this is more explicit (especially as you already have other return statements which return values)